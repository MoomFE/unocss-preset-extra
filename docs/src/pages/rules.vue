<template>
  <!-- 侧边栏 -->
  <BasicSidebar ref="sidebarRef" @setDrawerOpenBtn="(fn) => emit('setRenderNavbarLeft', fn)">
    <div class="w-60" un:flex="none">
      <div class="w-60 h-[calc(100%-64px)] fixed scrollbar" un:b-r="1 solid gray op-36">
        <n-menu v-model:value="value" :options="options" />
      </div>
    </div>
    <template #drawer>
      <n-menu v-model:value="value" :options="options" @update:value="sidebarRef.closeDrawer()" />
    </template>
  </BasicSidebar>

  <!-- 内容区域 -->
  <div class="px-6" un:p="x-6 t-5 b-10" un:flex="grow">
    <div class="text-3xl font-bold capitalize mb-1">{{ value }}</div>
    <router-view v-slot="{ Component, route }">
      <transition un:animated="~ faster" enter-active-class="animated-fade-in" leave-active-class="animated-fade-out" mode="out-in">
        <component :is="h('div', Component)" :key="route.path" />
      </transition>
    </router-view>
  </div>
</template>

<script lang="ts" setup>
  import type { MenuProps } from 'naive-ui';
  import { camelCase, upperFirst } from 'lodash';

  const emit = defineEmits(['setRenderNavbarLeft']);

  const route = useRoute();
  const router = useRouter();

  const sidebarRef = ref();

  /** 规则列表 */
  const rules = ['size', 'elevation', 'animated'];
  /** 菜单 */
  const options: MenuProps['options'] = rules.map(key => ({ label: key, key }));
  /** 当前选中的菜单项 */
  const value = ref(
    route.path.split('/').reverse()[0],
  );

  whenever(value, (key) => {
    router.push({
      name: `Rules/${upperFirst(camelCase(key))}`,
    });
  });
</script>

<route lang="yaml">
name: Rules
redirect:
  name: Rules/Size
meta:
  layout: home
  layoutContentClass: px-0!
</route>
