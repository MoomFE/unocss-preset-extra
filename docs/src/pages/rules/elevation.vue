<template>
  <div class="text-sm">
    <OuterChain href="https://vuetifyjs.com/zh-Hans/styles/elevation/" code>Vuetify 海拔</OuterChain> 样式
  </div>
  <n-divider />

  <n-h2>使用</n-h2>
  <DemoCard :code="codeStr">
    <div class="py-12">
      <ElevationDiv :index="index" :opacity="opacity" />
    </div>
    <template #options="{ showCode }">
      <n-form label-placement="left" label-width="4em" :show-feedback="false">
        <client-only>
          <n-form-item label="层级">
            <n-slider v-model:value="index" :max="24" />
          </n-form-item>
          <n-form-item label="透明度">
            <n-slider v-model:value="opacity" :max="100" />
          </n-form-item>
          <n-form-item v-if="showCode" label="简写">
            <n-checkbox v-model:checked="abbr" />
          </n-form-item>
        </client-only>
      </n-form>
    </template>
  </DemoCard>

  <n-h2>所有层级</n-h2>
  <n-card>
    <div un:flex="~ wrap gap-7.5">
      <ElevationDiv v-for="i in 25" :key="i" :index="i - 1" />
    </div>
  </n-card>
</template>

<script lang="tsx" setup>
  /** 层级 */
  const index = ref(6);
  /** 透明度 */
  const opacity = ref(100);
  /** 简写 */
  const abbr = ref(false);

  /** 代码字符串 */
  const codeStr = computed(() => {
    const el = abbr.value ? 'el' : 'elevation';
    const op = abbr.value ? 'op' : 'opacity';
    const opValue = opacity.value;

    return `<div class="${el}-${index.value}${opValue === 100 ? '' : ` ${el}-${op}-${opacity.value}`}" />`;
  });

  /** 海拔 */
  const ElevationDiv = ({ index, opacity }: { index: string | number; opacity?: number }) => (
    <div class={`size-25 flex justify-center items-center rounded c-white bg-sky el-${index} el-op-${opacity ?? 100}`}>{ index }</div>
  );
</script>

<route lang="yaml">
name: Rules/Elevation
</route>
