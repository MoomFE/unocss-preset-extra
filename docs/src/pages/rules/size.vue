<template>
  <div>size</div>
</template>

<script lang="ts" setup>

</script>

<route lang="yaml">
name: Rules/Size
</route>
