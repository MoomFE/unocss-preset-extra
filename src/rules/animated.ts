import { type CSSObject, type Rule } from 'unocss';
import { type Theme } from '@unocss/preset-mini';
import { handler } from '@unocss/preset-mini/utils';
import animatedJSON from './animated.json';

function getAnimated() {
  return animatedJSON as unknown as {
    [key: string]: {
      animationName: string
      css: CSSObject
      keyframes: string
    }
  };
}

export const durationShortcuts = {
  faster: 0.5,
  fast: 0.8,
  slow: 2,
  slower: 3,
};

/**
 * animate.css
 */
export const animatedRules: Rule<Theme>[] = [
  [
    /^animated$/,
    () => {
      return {
        '--une-animated-duration': '1s',
        'animation-duration': 'var(--une-animated-duration)',
        'animation-fill-mode': 'both',
      };
    },
    {
      autocomplete: ['animated'],
    },
  ],
  [
    new RegExp(`^animated-(${Object.keys(animatedJSON).join('|')})$`),
    ([, name]) => {
      const { animationName, css, keyframes } = getAnimated()[name];

      return [
        `@keyframes ${animationName} { ${keyframes} }`,
        css,
      ];
    },
    {
      autocomplete: [
        `animated-(${Object.keys(animatedJSON).join('|')})`,
      ],
    },
  ],
  [
    /^animated-(infinite|(repeat-(infinite|(\d+(\.\d+)?))))$/,
    ([,,, repeat]) => {
      const isInfinite = !repeat || repeat === 'infinite';

      return {
        'animation-iteration-count': isInfinite ? 'infinite' : repeat,
      };
    },
    {
      autocomplete: [
        'animated-infinite',
        'animated-repeat-infinite',
        'animated-repeat-<num>',
      ],
    },
  ],
  [
    /^animated-delay-(none|(\d+(\.\d+)?(m?s)?))$/,
    ([, d]) => ({
      'animation-delay': d === 'none' ? '0ms' : handler.bracket.cssvar.time(d),
    }),
    {
      autocomplete: [
        'animated-delay-none',
        'animated-delay-$duration',
      ],
    },
  ],
  [
    /^animated-(((fast|slow)(?:er)?)|duration-(none|(\d+(\.\d+)?(m?s)?)))/,
    ([_,, shortcut,, v]) => {
      if (shortcut) {
        return {
          'animation-duration': `calc(var(--une-animated-duration) * ${durationShortcuts[shortcut as keyof typeof durationShortcuts]});`,
        };
      }

      return {
        'animation-duration': v === 'none' ? '0ms' : handler.bracket.cssvar.time(v),
      };
    },
    {
      autocomplete: [
        `animated-(${Object.keys(durationShortcuts).join('|')})`,
        'animated-duration-none',
        'animated-duration-$duration',
      ],
    },
  ],
];
