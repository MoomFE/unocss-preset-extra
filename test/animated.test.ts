import { createGenerator, presetAttributify, presetUno } from 'unocss';
import { describe, expect, test } from 'vitest';
import { createAutocomplete } from '@unocss/autocomplete';
import postcss from 'postcss';
import postcssJs from 'postcss-js';
import { durationShortcuts } from '../src/rules/animated';
import animated from '../src/rules/animated.json';
import { removeLastZero } from './utils';
import { presetExtra } from '@/index';

describe('animated.json', () => {
  test('数据的 key 均为 kebabCase 格式', () => {
    Object.keys(animated).forEach((key) => {
      expect(/^[a-z-]+$/.test(key)).is.true;
    });
  });

  test('动画名称为 une 开头的 camelCase 格式', () => {
    Object.values(animated).forEach(({ animationName, css }) => {
      expect(/^une[A-Z]/.test(animationName)).is.true;
      expect(/^une[A-Z]/.test(css['animation-name'])).is.true;
    });
  });

  test('样式名为 kebabCase 格式', () => {
    Object.values(animated).forEach(({ css }) => {
      Object.keys(css).forEach((key) => {
        expect(/^[a-z-]+$/.test(key)).is.true;
      });
    });
  });
});

describe('animated', () => {
  const generator = createGenerator({
    presets: [
      presetUno(),
      presetAttributify(),
      presetExtra(),
    ],
  });

  test('animated', async () => {
    const { css } = await generator.generate('animated', { preflights: false });

    expect(
      postcssJs.objectify(postcss.parse(css)),
    ).toEqual({
      '.animated': {
        '--une-animated-duration': '1s',
        'animationDuration': 'var(--une-animated-duration)',
        'animationFillMode': 'both',
      },
    });
  });

  test('animated-name', async () => {
    const { css } = await generator.generate(
      Object.keys(animated).map(k => `animated-${k}`).join(' '),
      { preflights: false },
    );

    const styles = postcssJs.objectify(postcss.parse(css));

    // 均生成了 css 样式及 @keyframes
    Object.entries(animated).forEach(([key, { animationName }]) => {
      expect(styles[`.animated-${key}`]).toBeDefined();
      expect(styles[`@keyframes ${animationName}`]).toBeDefined();
    });
  });

  test('animated-repeat', async () => {
    const { css } = await generator.generate(`
      animated-infinite
      animated-repeat-infinite
      ${/* 0 ~ 66 */ Array.from({ length: 67 }, (_, i) => `animated-repeat-${i}`).join(' ')}
      ${/* 0.1, 1.2, ... ( 小数 ) */ Array.from({ length: 67 }, (_, i) => `animated-repeat-${i}.${removeLastZero(i + 1)}`).join(' ')}
      ${/* 0_1, 1_2, ... ( 不符合规则的样式类 ) */ Array.from({ length: 7 }, (_, i) => `animated-repeat-${i}_${removeLastZero(i + 1)}`).join(' ')}
      ${/* a ~ z ( 不符合规则的样式类 ) */ Array.from({ length: 26 }, (_, i) => `animated-repeat-${String.fromCharCode(97 + i)}`)}
    `, {
      preflights: false,
    });

    expect(
      postcssJs.objectify(postcss.parse(css)),
    ).toEqual(
      Object.assign(
        // infinite, repeat-infinite
        {
          '.animated-infinite,\n.animated-repeat-infinite': { animationIterationCount: 'infinite' },
        },
        // 0 ~ 66
        ...Array.from({ length: 67 }, (_, i) => ({
          [`.animated-repeat-${i}`]: { animationIterationCount: `${i}` },
        })),
        // 0.1, 1.2, ... ( 小数 )
        ...Array.from({ length: 67 }, (_, i) => ({
          [`.animated-repeat-${i}\\.${removeLastZero(i + 1)}`]: { animationIterationCount: `${i}.${removeLastZero(i + 1)}` },
        })),
      ),
    );
  });

  test('animated-delay', async () => {
    const { css } = await generator.generate(`
      animated-delay-none
      ${/* 0 ~ 66 */ Array.from({ length: 67 }, (_, i) => `animated-delay-${i}`).join(' ')}
      ${/* 0ms ~ 66ms */ Array.from({ length: 67 }, (_, i) => `animated-delay-${i}ms`).join(' ')}
      ${/* 0s ~ 66s */ Array.from({ length: 67 }, (_, i) => `animated-delay-${i}s`).join(' ')}
      ${/* 0.1, 1.2, ... ( 小数 ) */ Array.from({ length: 67 }, (_, i) => `animated-delay-${i}.${removeLastZero(i + 1)}`).join(' ')}
      ${/* 0.1ms, 1.2ms, ... ( 小数 ) */ Array.from({ length: 67 }, (_, i) => `animated-delay-${i}.${removeLastZero(i + 1)}ms`).join(' ')}
      ${/* 0.1s, 1.2s, ... ( 小数 ) */ Array.from({ length: 67 }, (_, i) => `animated-delay-${i}.${removeLastZero(i + 1)}s`).join(' ')}
    `, {
      preflights: false,
    });

    expect(css).toMatchSnapshot();
  });

  test('animated-duration', async () => {
    const { css } = await generator.generate(`
      animated-duration-none
      ${/* shortcuts */ Object.keys(durationShortcuts).map(k => `animated-${k}`).join(' ')}
      ${/* 0 ~ 66 */ Array.from({ length: 67 }, (_, i) => `animated-duration-${i}`).join(' ')}
      ${/* 0ms ~ 66ms */ Array.from({ length: 67 }, (_, i) => `animated-duration-${i}ms`).join(' ')}
      ${/* 0s ~ 66s */ Array.from({ length: 67 }, (_, i) => `animated-duration-${i}s`).join(' ')}
      ${/* 0.1, 1.2, ... ( 小数 ) */ Array.from({ length: 67 }, (_, i) => `animated-duration-${i}.${removeLastZero(i + 1)}`).join(' ')}
      ${/* 0.1ms, 1.2ms, ... ( 小数 ) */ Array.from({ length: 67 }, (_, i) => `animated-duration-${i}.${removeLastZero(i + 1)}ms`).join(' ')}
      ${/* 0.1s, 1.2s, ... ( 小数 ) */ Array.from({ length: 67 }, (_, i) => `animated-duration-${i}.${removeLastZero(i + 1)}s`).join(' ')}
    `, {
      preflights: false,
    });

    expect(css).toMatchSnapshot();
  });

  test('autocomplete', async () => {
    const autocomplete = createAutocomplete(createGenerator({
      presets: [
        presetUno(),
        presetAttributify(),
        presetExtra(),
      ],
    }));

    expect(await autocomplete.suggest('animated')).toMatchSnapshot();

    expect(await autocomplete.suggest('animated-')).toMatchSnapshot();

    expect(await autocomplete.suggest('animated-repeat-')).toMatchSnapshot();

    expect(await autocomplete.suggest('animated-delay-')).toMatchSnapshot();

    expect(await autocomplete.suggest('animated-duration-')).toMatchSnapshot();
  });
});
