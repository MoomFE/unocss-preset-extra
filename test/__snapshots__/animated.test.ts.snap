// Vitest Snapshot v1

exports[`animated > autocomplete 1`] = `
[
  "animated",
  "animated-delay-0",
  "animated-delay-0ms",
  "animated-delay-0s",
  "animated-delay-1",
  "animated-delay-2",
  "animated-delay-3",
  "animated-delay-4",
  "animated-delay-5",
  "animated-delay-6",
  "animated-delay-7",
  "animated-delay-8",
  "animated-delay-9",
  "animated-delay-10",
  "animated-delay-10ms",
  "animated-delay-10s",
  "animated-delay-11",
  "animated-delay-11ms",
  "animated-delay-11s",
  "animated-delay-12",
  "animated-delay-12ms",
  "animated-delay-12s",
  "animated-delay-13",
  "animated-delay-13ms",
  "animated-delay-13s",
  "animated-delay-14",
  "animated-delay-14ms",
  "animated-delay-14s",
  "animated-delay-15",
  "animated-delay-15ms",
  "animated-delay-15s",
  "animated-delay-16",
  "animated-delay-16ms",
  "animated-delay-16s",
  "animated-delay-17",
  "animated-delay-17ms",
  "animated-delay-17s",
  "animated-delay-18",
  "animated-delay-18ms",
  "animated-delay-18s",
  "animated-delay-19",
  "animated-delay-19ms",
  "animated-delay-19s",
  "animated-delay-1ms",
  "animated-delay-1s",
  "animated-delay-20",
  "animated-delay-20ms",
  "animated-delay-20s",
  "animated-delay-21",
  "animated-delay-21ms",
  "animated-delay-21s",
  "animated-delay-22",
  "animated-delay-22ms",
  "animated-delay-22s",
  "animated-delay-23",
  "animated-delay-23ms",
  "animated-delay-23s",
  "animated-delay-24",
  "animated-delay-24ms",
  "animated-delay-24s",
  "animated-delay-25",
  "animated-delay-25ms",
  "animated-delay-25s",
  "animated-delay-26",
  "animated-delay-26ms",
  "animated-delay-26s",
  "animated-delay-27",
  "animated-delay-27ms",
  "animated-delay-27s",
  "animated-delay-28",
  "animated-delay-28ms",
  "animated-delay-28s",
  "animated-delay-29",
  "animated-delay-29ms",
  "animated-delay-29s",
  "animated-delay-2ms",
  "animated-delay-2s",
  "animated-delay-30",
  "animated-delay-30ms",
  "animated-delay-30s",
  "animated-delay-31",
  "animated-delay-31ms",
  "animated-delay-31s",
  "animated-delay-32",
  "animated-delay-32ms",
  "animated-delay-32s",
  "animated-delay-33",
  "animated-delay-33ms",
  "animated-delay-33s",
  "animated-delay-34",
  "animated-delay-34ms",
  "animated-delay-34s",
  "animated-delay-35",
  "animated-delay-35ms",
  "animated-delay-35s",
  "animated-delay-36",
  "animated-delay-36ms",
  "animated-delay-36s",
  "animated-delay-37",
  "animated-delay-37ms",
  "animated-delay-37s",
  "animated-delay-38",
  "animated-delay-38ms",
  "animated-delay-38s",
  "animated-delay-39",
  "animated-delay-39ms",
  "animated-delay-39s",
  "animated-delay-3ms",
  "animated-delay-3s",
  "animated-delay-40",
  "animated-delay-40ms",
  "animated-delay-40s",
  "animated-delay-41",
  "animated-delay-41ms",
  "animated-delay-41s",
  "animated-delay-42",
  "animated-delay-42ms",
  "animated-delay-42s",
  "animated-delay-43",
  "animated-delay-43ms",
  "animated-delay-43s",
  "animated-delay-44",
  "animated-delay-44ms",
  "animated-delay-44s",
  "animated-delay-45",
  "animated-delay-45ms",
  "animated-delay-45s",
  "animated-delay-46",
  "animated-delay-46ms",
  "animated-delay-46s",
  "animated-delay-47",
  "animated-delay-47ms",
  "animated-delay-47s",
  "animated-delay-48",
  "animated-delay-48ms",
  "animated-delay-48s",
  "animated-delay-49",
  "animated-delay-49ms",
  "animated-delay-49s",
  "animated-delay-4ms",
  "animated-delay-4s",
  "animated-delay-50",
  "animated-delay-50ms",
  "animated-delay-50s",
  "animated-delay-51",
  "animated-delay-51ms",
  "animated-delay-51s",
  "animated-delay-52",
  "animated-delay-52ms",
  "animated-delay-52s",
  "animated-delay-53",
  "animated-delay-53ms",
  "animated-delay-53s",
  "animated-delay-54",
  "animated-delay-54ms",
  "animated-delay-54s",
  "animated-delay-55",
  "animated-delay-55ms",
  "animated-delay-55s",
  "animated-delay-56",
  "animated-delay-56ms",
  "animated-delay-56s",
  "animated-delay-57",
  "animated-delay-57ms",
  "animated-delay-57s",
  "animated-delay-58",
  "animated-delay-58ms",
  "animated-delay-58s",
  "animated-delay-59",
  "animated-delay-59ms",
  "animated-delay-59s",
  "animated-delay-5ms",
  "animated-delay-5s",
  "animated-delay-60",
  "animated-delay-60ms",
  "animated-delay-60s",
  "animated-delay-61",
  "animated-delay-61ms",
  "animated-delay-61s",
  "animated-delay-62",
  "animated-delay-62ms",
  "animated-delay-62s",
  "animated-delay-63",
  "animated-delay-63ms",
  "animated-delay-63s",
  "animated-delay-64",
  "animated-delay-64ms",
  "animated-delay-64s",
  "animated-delay-65",
  "animated-delay-65ms",
  "animated-delay-65s",
  "animated-delay-66",
  "animated-delay-66ms",
  "animated-delay-66s",
  "animated-delay-6ms",
  "animated-delay-6s",
  "animated-delay-7ms",
  "animated-delay-7s",
  "animated-delay-8ms",
  "animated-delay-8s",
  "animated-delay-9ms",
  "animated-delay-9s",
  "animated-delay-none",
  "animated-infinite",
  "animated-repeat-0",
  "animated-repeat-1",
  "animated-repeat-0.1",
  "animated-repeat-2",
  "animated-repeat-1.2",
  "animated-repeat-3",
  "animated-repeat-2.3",
  "animated-repeat-4",
  "animated-repeat-3.4",
  "animated-repeat-5",
  "animated-repeat-4.5",
  "animated-repeat-6",
  "animated-repeat-5.6",
  "animated-repeat-7",
  "animated-repeat-6.7",
  "animated-repeat-8",
  "animated-repeat-7.8",
  "animated-repeat-9",
  "animated-repeat-8.9",
  "animated-repeat-10",
  "animated-repeat-9.10",
  "animated-repeat-11",
  "animated-repeat-10.11",
  "animated-repeat-12",
  "animated-repeat-11.12",
  "animated-repeat-13",
  "animated-repeat-12.13",
  "animated-repeat-14",
  "animated-repeat-13.14",
  "animated-repeat-15",
  "animated-repeat-14.15",
  "animated-repeat-16",
  "animated-repeat-15.16",
  "animated-repeat-17",
  "animated-repeat-16.17",
  "animated-repeat-18",
  "animated-repeat-17.18",
  "animated-repeat-19",
  "animated-repeat-18.19",
  "animated-repeat-20",
  "animated-repeat-19.20",
  "animated-repeat-21",
  "animated-repeat-20.21",
  "animated-repeat-22",
  "animated-repeat-21.22",
  "animated-repeat-23",
  "animated-repeat-22.23",
  "animated-repeat-24",
  "animated-repeat-23.24",
  "animated-repeat-25",
  "animated-repeat-24.25",
  "animated-repeat-26",
  "animated-repeat-25.26",
  "animated-repeat-27",
  "animated-repeat-26.27",
  "animated-repeat-28",
  "animated-repeat-27.28",
  "animated-repeat-29",
  "animated-repeat-28.29",
  "animated-repeat-30",
  "animated-repeat-29.30",
  "animated-repeat-31",
  "animated-repeat-30.31",
  "animated-repeat-32",
  "animated-repeat-31.32",
  "animated-repeat-33",
  "animated-repeat-32.33",
  "animated-repeat-34",
  "animated-repeat-33.34",
  "animated-repeat-35",
  "animated-repeat-34.35",
  "animated-repeat-36",
  "animated-repeat-35.36",
  "animated-repeat-37",
  "animated-repeat-36.37",
  "animated-repeat-38",
  "animated-repeat-37.38",
  "animated-repeat-39",
  "animated-repeat-38.39",
  "animated-repeat-40",
  "animated-repeat-39.40",
  "animated-repeat-41",
  "animated-repeat-40.41",
  "animated-repeat-42",
  "animated-repeat-41.42",
  "animated-repeat-43",
  "animated-repeat-42.43",
  "animated-repeat-44",
  "animated-repeat-43.44",
  "animated-repeat-45",
  "animated-repeat-44.45",
  "animated-repeat-46",
  "animated-repeat-45.46",
  "animated-repeat-47",
  "animated-repeat-46.47",
  "animated-repeat-48",
  "animated-repeat-47.48",
  "animated-repeat-49",
  "animated-repeat-48.49",
  "animated-repeat-50",
  "animated-repeat-49.50",
  "animated-repeat-51",
  "animated-repeat-50.51",
  "animated-repeat-52",
  "animated-repeat-51.52",
  "animated-repeat-53",
  "animated-repeat-52.53",
  "animated-repeat-54",
  "animated-repeat-53.54",
  "animated-repeat-55",
  "animated-repeat-54.55",
  "animated-repeat-56",
  "animated-repeat-55.56",
  "animated-repeat-57",
  "animated-repeat-56.57",
  "animated-repeat-58",
  "animated-repeat-57.58",
  "animated-repeat-59",
  "animated-repeat-58.59",
  "animated-repeat-60",
  "animated-repeat-59.60",
  "animated-repeat-61",
  "animated-repeat-60.61",
  "animated-repeat-62",
  "animated-repeat-61.62",
  "animated-repeat-63",
  "animated-repeat-62.63",
  "animated-repeat-64",
  "animated-repeat-63.64",
  "animated-repeat-65",
  "animated-repeat-64.65",
  "animated-repeat-66",
  "animated-repeat-65.66",
  "animated-repeat-66.67",
  "animated-repeat-infinite",
]
`;

exports[`animated > autocomplete 2`] = `
[
  "animated-repeat-0",
  "animated-repeat-1",
  "animated-repeat-0.1",
  "animated-repeat-2",
  "animated-repeat-1.2",
  "animated-repeat-3",
  "animated-repeat-2.3",
  "animated-repeat-4",
  "animated-repeat-3.4",
  "animated-repeat-5",
  "animated-repeat-4.5",
  "animated-repeat-6",
  "animated-repeat-5.6",
  "animated-repeat-7",
  "animated-repeat-6.7",
  "animated-repeat-8",
  "animated-repeat-7.8",
  "animated-repeat-9",
  "animated-repeat-8.9",
  "animated-repeat-10",
  "animated-repeat-9.10",
  "animated-repeat-11",
  "animated-repeat-10.11",
  "animated-repeat-12",
  "animated-repeat-11.12",
  "animated-repeat-13",
  "animated-repeat-12.13",
  "animated-repeat-14",
  "animated-repeat-13.14",
  "animated-repeat-15",
  "animated-repeat-14.15",
  "animated-repeat-16",
  "animated-repeat-15.16",
  "animated-repeat-17",
  "animated-repeat-16.17",
  "animated-repeat-18",
  "animated-repeat-17.18",
  "animated-repeat-19",
  "animated-repeat-18.19",
  "animated-repeat-20",
  "animated-repeat-19.20",
  "animated-repeat-21",
  "animated-repeat-20.21",
  "animated-repeat-22",
  "animated-repeat-21.22",
  "animated-repeat-23",
  "animated-repeat-22.23",
  "animated-repeat-24",
  "animated-repeat-23.24",
  "animated-repeat-25",
  "animated-repeat-24.25",
  "animated-repeat-26",
  "animated-repeat-25.26",
  "animated-repeat-27",
  "animated-repeat-26.27",
  "animated-repeat-28",
  "animated-repeat-27.28",
  "animated-repeat-29",
  "animated-repeat-28.29",
  "animated-repeat-30",
  "animated-repeat-29.30",
  "animated-repeat-31",
  "animated-repeat-30.31",
  "animated-repeat-32",
  "animated-repeat-31.32",
  "animated-repeat-33",
  "animated-repeat-32.33",
  "animated-repeat-34",
  "animated-repeat-33.34",
  "animated-repeat-35",
  "animated-repeat-34.35",
  "animated-repeat-36",
  "animated-repeat-35.36",
  "animated-repeat-37",
  "animated-repeat-36.37",
  "animated-repeat-38",
  "animated-repeat-37.38",
  "animated-repeat-39",
  "animated-repeat-38.39",
  "animated-repeat-40",
  "animated-repeat-39.40",
  "animated-repeat-41",
  "animated-repeat-40.41",
  "animated-repeat-42",
  "animated-repeat-41.42",
  "animated-repeat-43",
  "animated-repeat-42.43",
  "animated-repeat-44",
  "animated-repeat-43.44",
  "animated-repeat-45",
  "animated-repeat-44.45",
  "animated-repeat-46",
  "animated-repeat-45.46",
  "animated-repeat-47",
  "animated-repeat-46.47",
  "animated-repeat-48",
  "animated-repeat-47.48",
  "animated-repeat-49",
  "animated-repeat-48.49",
  "animated-repeat-50",
  "animated-repeat-49.50",
  "animated-repeat-51",
  "animated-repeat-50.51",
  "animated-repeat-52",
  "animated-repeat-51.52",
  "animated-repeat-53",
  "animated-repeat-52.53",
  "animated-repeat-54",
  "animated-repeat-53.54",
  "animated-repeat-55",
  "animated-repeat-54.55",
  "animated-repeat-56",
  "animated-repeat-55.56",
  "animated-repeat-57",
  "animated-repeat-56.57",
  "animated-repeat-58",
  "animated-repeat-57.58",
  "animated-repeat-59",
  "animated-repeat-58.59",
  "animated-repeat-60",
  "animated-repeat-59.60",
  "animated-repeat-61",
  "animated-repeat-60.61",
  "animated-repeat-62",
  "animated-repeat-61.62",
  "animated-repeat-63",
  "animated-repeat-62.63",
  "animated-repeat-64",
  "animated-repeat-63.64",
  "animated-repeat-65",
  "animated-repeat-64.65",
  "animated-repeat-66",
  "animated-repeat-65.66",
  "animated-repeat-66.67",
  "animated-repeat-infinite",
]
`;

exports[`animated > autocomplete 3`] = `
[
  "animated-delay-0",
  "animated-delay-0ms",
  "animated-delay-0s",
  "animated-delay-1",
  "animated-delay-10ms",
  "animated-delay-10s",
  "animated-delay-11ms",
  "animated-delay-11s",
  "animated-delay-12ms",
  "animated-delay-12s",
  "animated-delay-13ms",
  "animated-delay-13s",
  "animated-delay-14ms",
  "animated-delay-14s",
  "animated-delay-15ms",
  "animated-delay-15s",
  "animated-delay-16ms",
  "animated-delay-16s",
  "animated-delay-17ms",
  "animated-delay-17s",
  "animated-delay-18ms",
  "animated-delay-18s",
  "animated-delay-19ms",
  "animated-delay-19s",
  "animated-delay-1ms",
  "animated-delay-1s",
  "animated-delay-2",
  "animated-delay-20ms",
  "animated-delay-20s",
  "animated-delay-21ms",
  "animated-delay-21s",
  "animated-delay-22ms",
  "animated-delay-22s",
  "animated-delay-23ms",
  "animated-delay-23s",
  "animated-delay-24ms",
  "animated-delay-24s",
  "animated-delay-25ms",
  "animated-delay-25s",
  "animated-delay-26ms",
  "animated-delay-26s",
  "animated-delay-27ms",
  "animated-delay-27s",
  "animated-delay-28ms",
  "animated-delay-28s",
  "animated-delay-29ms",
  "animated-delay-29s",
  "animated-delay-2ms",
  "animated-delay-2s",
  "animated-delay-3",
  "animated-delay-30ms",
  "animated-delay-30s",
  "animated-delay-31ms",
  "animated-delay-31s",
  "animated-delay-32ms",
  "animated-delay-32s",
  "animated-delay-33ms",
  "animated-delay-33s",
  "animated-delay-34ms",
  "animated-delay-34s",
  "animated-delay-35ms",
  "animated-delay-35s",
  "animated-delay-36ms",
  "animated-delay-36s",
  "animated-delay-37ms",
  "animated-delay-37s",
  "animated-delay-38ms",
  "animated-delay-38s",
  "animated-delay-39ms",
  "animated-delay-39s",
  "animated-delay-3ms",
  "animated-delay-3s",
  "animated-delay-4",
  "animated-delay-40ms",
  "animated-delay-40s",
  "animated-delay-41ms",
  "animated-delay-41s",
  "animated-delay-42ms",
  "animated-delay-42s",
  "animated-delay-43ms",
  "animated-delay-43s",
  "animated-delay-44ms",
  "animated-delay-44s",
  "animated-delay-45ms",
  "animated-delay-45s",
  "animated-delay-46ms",
  "animated-delay-46s",
  "animated-delay-47ms",
  "animated-delay-47s",
  "animated-delay-48ms",
  "animated-delay-48s",
  "animated-delay-49ms",
  "animated-delay-49s",
  "animated-delay-4ms",
  "animated-delay-4s",
  "animated-delay-5",
  "animated-delay-50ms",
  "animated-delay-50s",
  "animated-delay-51ms",
  "animated-delay-51s",
  "animated-delay-52ms",
  "animated-delay-52s",
  "animated-delay-53ms",
  "animated-delay-53s",
  "animated-delay-54ms",
  "animated-delay-54s",
  "animated-delay-55ms",
  "animated-delay-55s",
  "animated-delay-56ms",
  "animated-delay-56s",
  "animated-delay-57ms",
  "animated-delay-57s",
  "animated-delay-58ms",
  "animated-delay-58s",
  "animated-delay-59ms",
  "animated-delay-59s",
  "animated-delay-5s",
  "animated-delay-6",
  "animated-delay-60ms",
  "animated-delay-60s",
  "animated-delay-61ms",
  "animated-delay-61s",
  "animated-delay-62ms",
  "animated-delay-62s",
  "animated-delay-63ms",
  "animated-delay-63s",
  "animated-delay-64ms",
  "animated-delay-64s",
  "animated-delay-65ms",
  "animated-delay-65s",
  "animated-delay-66ms",
  "animated-delay-66s",
  "animated-delay-6s",
  "animated-delay-7",
  "animated-delay-8",
  "animated-delay-9",
  "animated-delay-10",
  "animated-delay-11",
  "animated-delay-12",
  "animated-delay-13",
  "animated-delay-14",
  "animated-delay-15",
  "animated-delay-16",
  "animated-delay-17",
  "animated-delay-18",
  "animated-delay-19",
  "animated-delay-20",
  "animated-delay-21",
  "animated-delay-22",
  "animated-delay-23",
  "animated-delay-24",
  "animated-delay-25",
  "animated-delay-26",
  "animated-delay-27",
  "animated-delay-28",
  "animated-delay-29",
  "animated-delay-30",
  "animated-delay-31",
  "animated-delay-32",
  "animated-delay-33",
  "animated-delay-34",
  "animated-delay-35",
  "animated-delay-36",
  "animated-delay-37",
  "animated-delay-38",
  "animated-delay-39",
  "animated-delay-40",
  "animated-delay-41",
  "animated-delay-42",
  "animated-delay-43",
  "animated-delay-44",
  "animated-delay-45",
  "animated-delay-46",
  "animated-delay-47",
  "animated-delay-48",
  "animated-delay-49",
  "animated-delay-50",
  "animated-delay-51",
  "animated-delay-52",
  "animated-delay-53",
  "animated-delay-54",
  "animated-delay-55",
  "animated-delay-56",
  "animated-delay-57",
  "animated-delay-58",
  "animated-delay-59",
  "animated-delay-5ms",
  "animated-delay-60",
  "animated-delay-61",
  "animated-delay-62",
  "animated-delay-63",
  "animated-delay-64",
  "animated-delay-65",
  "animated-delay-66",
  "animated-delay-6ms",
  "animated-delay-75",
  "animated-delay-100",
  "animated-delay-150",
  "animated-delay-200",
  "animated-delay-300",
  "animated-delay-500",
  "animated-delay-700",
  "animated-delay-1000",
  "animated-delay-7ms",
  "animated-delay-7s",
  "animated-delay-8ms",
  "animated-delay-8s",
  "animated-delay-9ms",
  "animated-delay-9s",
  "animated-delay-none",
]
`;
